File: js/7kyu/alphabet_war.js
// https://www.codewars.com/kata/59377c53e66267c8f6000027/train/javascript

function alphabetWar(fight)
{
  let leftDict = {"w": 4, "p": 3, "b": 2, "s": 1};
  let rigthDict = {"m": 4, "q": 3, "d": 2, "z": 1};

  let leftScore = 0;
  let rightScore = 0;

  // could have use fight.split('')
  // or : for char in fight 
  Array.from(fight).forEach((char) => {
    if (char in leftDict) {
      leftScore += leftDict[char]
    }
    if (char in rigthDict) {
      rightScore += rigthDict[char]
    }
  });


  if (leftScore > rightScore) {
    return "Left side wins!"
  }
  if (rightScore > leftScore) {
    return "Right side wins!"
  }
  return "Let's fight again!";
}

File: python/7kyu/alphabet_war.py
def alphabet_war(fight):
  left_dict = dict(zip(['w','p','b','s'],[4,3,2,1]))
  right_dict = dict(zip(['m','q','d','z'],[4,3,2,1]))

  left_score = 0
  right_score = 0

  for char in fight:
    if (char in left_dict):
      left_score += left_dict[char]
    elif (char in right_dict):
      right_score += right_dict[char]
    else:
      pass

  if (left_score > right_score):
    return 'Left side wins!'
  elif (left_score == right_score):
    return "Let's fight again!"
  else:
    return 'Right side wins!'